lwt/tls_lwt.ml

```diff
@@ -276,11 +322,7 @@ type oc = Lwt_io.output_channel
 let of_t ?close t =
   let close = match close with
     | Some f -> (fun () -> Unix.safely (f ()))
-    | None   -> (fun () ->
-        (* (partially) avoid double-closes by checking if the fd has already been closed *)
-        match Lwt_unix.state t.Unix.fd with
-        | Lwt_unix.Closed -> Lwt.return_unit
-        | Lwt_unix.Opened | Lwt_unix.Aborted _ -> Unix.(safely (close t)))
+    | None   -> (fun () -> Unix.(safely (close t)))
```

Comment:
sorry I am puzzled why this change is part of this PR? And is the change intentional? If yes, please elaborate.

Reply:
ah, it moved into the Lwt_fd module, and inside of Unix.t we now have a Lwt_fd.t